Test/git-issues/git-issue-495.dfy.expect

```diff
@@ -1,17 +1,17 @@
-git-issue-495.dfy(9,12): Error: value does not satisfy the subset constraints of 'nat'
+git-issue-495.dfy(7,8): Error: value does not satisfy the subset constraints of 'nat'
```

Comment:
I would argue (9,12) is better error location than (7,8). Is it possible to get (9,12) instead?

Reply:
This was due to the patterns generated during pattern-match compilation possibly standing in for multiple branches (the token of the selector is used for those). I've now made reported error locations a bit more precise now by keeping track of the token of the original branch (in each RBranch), and using that in place of the selector's token when generated variables are bound in that branch.